### PR TITLE
Refactor/remove nist api

### PR DESCRIPTION
**Pull request by ionut-cmd**
Removed NIST API dependency completely.

I've had to change the `getHistoricalCves` function to `getCVEInfo`, as we do not have any historical CVE data + all the function was returning previously is multiple CVES with the exact name and not much info. It was not used in any other part of the code.

Now we have a new function instead that returns some more meaningful information like:
nqmvul -getCveInfo "CVE-2022-48174"                                  
    Fetching historical CVEs from API
 ```js
{
    cve: 'CVE-2022-48174',
    description: 'There is a stack overflow vulnerability in ash.c:6030 in busybox before 1.35. In the environment of Internet of Vehicles, this vulnerability can be executed from command to arbitrary code execution.',
    baseScore: 9.8,
    impactScore: 5.9,
    exploitabilityScore: 3.9,
    publishedDate: '2023-08-22T19:16Z',
    lastModifiedDate: '2024-11-29T12:15Z',
    cveDataVersion: '4.0'
  }
```

I still have to:

- Locate and remove the local database dependencies from other parts of the code. (some functions are still writing CPE data locally, we no longer need this as we now have a centralized DB) from other parts of the code.
- Redo all tests
- Update docs

For now I will focus on deploying the api before continuing with the code refactor, This will make testing

### CODE REVIEW

**Comment by ionut-cmd** [comment]
To test the sbom gap tool you will have to:

1. add a .env file in `packages/sbom-gap-cli/nist-data-api/.env` (details sent on watt) - this is for the PG database running on the dell server

2. run npm start in `packages/sbom-gap-cli/nist-data-api ` to start the server

File: packages/sbom-gap-cli/config/config.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "cpeBaseUrl": "http://192.168.1.111:3003/api/cpe-versions",
+    "cveHistUrl": "http://192.168.1.111:3003/api/cve-id",
+    "cveBaseUrl": "http://192.168.1.111:3003/api/cve",
+    "gitAdvisoryDbPath": "/home/ionut/Repositories/advisory-database/advisories"
+  }
```
Comment (anthonymccaigue) [inline packages/sbom-gap-cli/config/config.json:6]: add to gitignore?

**Review by anthonymccaigue — APPROVED** [review]
Looks GREAT! I just added a few items to package.json

File: packages/sbom-gap-cli/config/config.json (hunk 1 above)
Comment (ionut-cmd) [reply packages/sbom-gap-cli/config/config.json:6]: Yep, that's a good ideea

**Pull request merged**